pkg/state/state.go

```diff
@@ -2573,11 +2573,28 @@ func (st *HelmState) appendExtraSyncFlags(flags []string, opt *SyncOpts) []strin
 	return flags
 }
 
+// appendVerifyFlags append the --verify flags related to verify
+func (st *HelmState) appendVerifyFlags(flags []string, release *ReleaseSpec) []string {
+	repo, _ := st.GetRepositoryAndNameFromChartName(release.Chart)
+	switch {
+	case release.Verify != nil && *release.Verify:
+		flags = append(flags, "--verify")
+	case repo != nil && repo.Verify:
+		flags = append(flags, "--verify")
+	case st.HelmDefaults.Verify:
+		flags = append(flags, "--verify")
+	}
+	return flags
+}
+
 // appendKeyringFlags append all the helm command-line flags related to keyring
 func (st *HelmState) appendKeyringFlags(flags []string, release *ReleaseSpec) []string {
+	repo, _ := st.GetRepositoryAndNameFromChartName(release.Chart)
 	switch {
 	case release.Keyring != "":
 		flags = append(flags, "--keyring", release.Keyring)
+	case repo != nil && repo.Keyring != "":
```

Comment:
Correct me if I'm wrong, but I think this case is redundant because L2594 says it appends a `--keyring` flag when `release.Keyring != ""`, regardless of if `repo` is nil or not!

Reply:
when `release.Keyring == ""`, I think that we should follow the `repo` configration.

Reply:
@mumoshu

Reply:
Ahh, gotcha! release.Keyring vs repo.Keyring! Thanks for the explanation!